enclave/core/switchlesscalls.c

```diff
@@ -71,9 +89,21 @@ oe_result_t oe_handle_init_switchless(uint64_t arg_in)
     // Copy the worker context array pointer and its size to avoid TOCTOU
     _host_worker_count = safe_manager.num_host_workers;
     _host_worker_contexts = safe_manager.host_worker_contexts;
+
+    // Check if already initialized
+    if (!oe_atomic_compare_and_swap(
+            (volatile int64_t*)&_is_switchless_initialized,
+            (int64_t) false,
+            (int64_t) true))
+    {
+        OE_RAISE(OE_ALREADY_INITIALIZED);
+    }
+
     result = OE_OK;
 
 done:
+    __atomic_store_n(&_switchless_init_in_progress, false, __ATOMIC_SEQ_CST);
```

Comment:
@jhand2 First, thanks for taking the suggestions and improving the PR.

When __switchless_init_in_progress_ is reset, it will allow _oe_handle_init_switchless_ to be called again.

Consider renaming this flag to __switchless_init_called_ and not clear this flag. That will match the PR description in a thread-safe manner.

Reply:
I'll make a call to oe_is_switchless_initialized instead. This gives better return code semantics.

Reply:
That would work too.
Consider moving clearing of `in progress` flag to before done to better reflect the initialization semantics.